strawberryfields/backends/bosonicbackend/bosoniccircuit.py

```diff
@@ -0,0 +1,833 @@
+# Copyright 2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Bosonic circuit operations"""
+# pylint: disable=duplicate-code,attribute-defined-outside-init
+import itertools as it
+import numpy as np
+from scipy.linalg import block_diag
+import thewalrus.symplectic as symp
+
+from strawberryfields.backends.bosonicbackend import ops
+
+
+# Shape of the weights, means, and covs arrays.
+def w_shape(num_modes, num_gauss):
+    r"""Calculate total number of weights. Assumes same number of weights per mode.
```

Comment:
(Minor): some of the docstrings might not necessarily need to be raw strings (no special characters)
```suggestion
    """Calculate total number of weights. Assumes same number of weights per mode.
```